src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -260,7 +260,9 @@
  * <li>each member layout in {@code G.memberLayouts()} is either a padding layout or
  *     a layout supported by {@code NL}, and</li>
  * <li>{@code G} does not contain padding other than what is strictly required to align
- *      its non-padding layout elements, or to satisfy (2).</li>
+ *      its non-padding layout elements, or to satisfy (2), and</li>
+ * <li>{@code G} is {@code G.memberLayouts().isEmpty()} or (at the same time) not all
+ * elements are padding layouts.</li>
```

Comment:
I suggest splitting the last list item into 2. The prior item also needs to be cleaned up (replace `, and` with `;`).

Not sure what you mean here with 'at the same time'? All of the items in the list should be true.

Reply:
I think it might be enough to just mention that not only padding layouts are allowed. The list is for things that should all be true, so adding an 'or' condition in there seems a bit confusing.